airqo-platform/AirQo-api
# fetching measurements from BQ

**_WHAT DOES THIS PR DO?_**

- [x] creating util function for fetching measurements from BQ
- [x] integrate the util function with the Get Events endpoint, V2
- [ ] Ensure that the input validations for datetime queries are exhaustive. They include:

- startTime
- endTime

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [PLAT-1154]

**_HOW DO I TEST OUT THIS PR?_**
First ensure that you have your _.env_  and other security files available and up-to-date. [More details are in Slack](https://airqo.slack.com/archives/C03218Z31JT/p1655974243775249).
```
cd src/device-registry
```
for Linux
`npm run dev-mac`

for Windows
`npm run dev-pc`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [Version 2 Get Events](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/device-registry/events#version-2-get-events)

**_SCREENSHOTS?:_** 
<img width="793" alt="Screenshot 2022-06-23 at 11 09 47" src="https://user-images.githubusercontent.com/1590213/175262806-897378d7-48f7-4910-92ff-101675f35a55.png">

<img width="789" alt="Screenshot 2022-06-25 at 21 03 01" src="https://user-images.githubusercontent.com/1590213/175787222-9414b543-7e05-462c-8979-264ae717346e.png">




[PLAT-1154]: https://airqoteam.atlassian.net/browse/PLAT-1154?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
